src/measurement_buffer.cpp

```diff
@@ -173,9 +176,9 @@ void MeasurementBuffer::BufferROSCloud(const sensor_msgs::PointCloud2& cloud)
       pass_through_filter.setFilterLimits( \
                   _min_obstacle_height,_max_obstacle_height);
       pass_through_filter.filter(*cloud_filtered);
+      pcl_conversions::fromPCL(*cloud_filtered, *cld_global);
```

Comment:
Seems like you should move the `    pcl_conversions::toPCL(*cld_global, *cloud_pcl);` line into both the if/else statements if no longer requires either. That conversion is non-trivial in time

Reply:
Sorry if I'm extra slow today, but I'm not sure what you mean. I thought that's what I did. That line only runs if one of the filters is active, else it's skipped. The only `pcl_conversions::fromPCL` in all of STVL seem to be on measurement_buffer.cpp , both in those blocks. Could you please elaborate a bit?

Reply:
Excellent suggestion!